core/src/main/java/org/mapstruct/TargetProperty.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks a <em>presence check method</em> parameter as a property name parameter.
+ * <p>
+ *   This parameter enables conditional filtering based on source property name in run-time.
+ * </p>
+ * @author Nikola Ivačič.
+ * @since 1.5
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.CLASS)
+public @interface TargetProperty {
```

Comment:
I would perhaps opt for a slightly longer name: `TargetPropertyName`. This leaves an option open to have a `TargetPropertyValue` (e.g. a Supplier function)

Reply:
That was my initial implementation but it was suggested to me to rename it to `@TargetProperty` :D.

Reply:
Indeed, me and @Zegveld suggested `@TargetProperty` initially. We have `@TargetType` for the `Class` of the type. In any case, we can always easily change this before the merge if we thing `@TagetProperty` is a better name (I think that @ivlcic changed it back to `@TargetPropertyName`)

Reply:
Ok.. given your example: the `Type` in `TargetType` refers to `class` or `interface`. So `Name` in `@TargetPropertyName` refers to a text or - in other words - `String`.  It leaves the option open for extension towards the future. I'm not bothered by the non-verbosity, since a proper IDE completes it anyway..

@filiphr : Enough on my side of the argument. I leave it up for you tot decide 😄 .

Reply:
Thinking out of the box here, but if someone wants to manage all their properties using an `enum` or some other container. Why wouldn't we allow this as long as we can convert to that type? It does open up some nice management options for people who want to use that route.

The above suggestion should be a separate PR though, but at that point having it already listed as `@TargetProperty` might be more useful. Although I can understand changing it to `@TargetPropertyName` because this is more verbose and still to the point even with the above suggestion.

I'm fine with either annotation.